src/auth/strategies/util.ts

```diff
@@ -25,6 +25,12 @@ export async function ExternalServiceCallback(
     // If `user` exists, the user has already logged in with this service and is good-to-go
     let user = await User.findOne({ [`services.${serviceName}.id`]: id });
 
+    // Ensure that the oauth method used is linked to the user with the same email address that is entered
+    if (session.email && user && session.email !== user.email) {
+        done(null, false, { message: "Login method is linked to another email. Please try again with a different email address." });
```

Comment:
This message is kind of confusing imo. Perhaps `The 3rd party account you chose does not match the email address that you are trying to log in with. Please make sure your email address and the selected account match.`

Reply:
Adding to this, users will be confused by this if we aren't really clear.  So it would be good to include the original email the user entered in the error message to.  As in, if I enter someone@example.com on the initial login screen, then choose a Google account with the email someone2@example.com, the error message might show, "The Google account you signed in with (someone2@example.com) has a different email than what you originally entered (someone@example.com).  Please ensure your email address and the account you select have the same email address.

Reply:
If someone uses their gt email, user@gatech.edu, but chooses to sign in with google (user@google.com), and on the first screen they enter their google email, how should the error message read?